flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -373,20 +375,30 @@ protected void addToSelectList(
         final SqlOperator operator = call.getOperator();
 
         if (operator instanceof SqlWindowTableFunction) {
-            if (explicitTableArgs.stream().allMatch(Objects::isNull)) {
+            if (tableArgs.stream().allMatch(Objects::isNull)) {
                 return rewritten;
             }
 
+            final boolean isSessionWindow = isSessionWindow(operator);
             final List<SqlIdentifier> descriptors =
                     call.getOperandList().stream()
                             .flatMap(FlinkCalciteSqlValidator::extractDescriptors)
                             .collect(Collectors.toList());
 
             for (int i = 0; i < call.operandCount(); i++) {
-                final SqlIdentifier tableArg = explicitTableArgs.get(i);
+                final SqlIdentifier tableArg = tableArgs.get(i);
                 if (tableArg != null) {
                     final SqlNode opReplacement = new ExplicitTableSqlSelect(tableArg, descriptors);
-                    if (call.operand(i).getKind() == SqlKind.ARGUMENT_ASSIGNMENT) {
+                    if (isSessionWindow) {
```

Comment:
The logic should be independent of any specific window function. This is why there was no explicit check for a TUMBLE, HOP, or CUMULATE. Just adding an additional branch for `else if (call.operand(i).getKind() == SqlKind.SET_SEMANTICS_TABLE)` should do the job, no? It should apply to all functions that with SET_SEMANTICS_TABLE args.

Reply:
>else if (call.operand(i).getKind() == SqlKind.SET_SEMANTICS_TABLE) should do the job, no?

in case of  `SET_SEMANTICS_TABLES` yes
however there is still `SqlKind.ARGUMENT_ASSIGNMENT` which should be handled differently because of how sqlSetSemanticsTableOperator looks like 
https://github.com/apache/flink/blob/0e373b20d917a9bcd2b00e32d3ff09b353aef728/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java#L396-L400
vs
https://github.com/apache/flink/blob/0e373b20d917a9bcd2b00e32d3ff09b353aef728/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java#L401-L405

Reply:
otherwise SESSION window will lose `PARTITION BY` in plans
as covered by
https://github.com/apache/flink/blob/0e373b20d917a9bcd2b00e32d3ff09b353aef728/flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala#L293-L305

Reply:
after looking once more i think in assignment we could also check kind op operator under it and apply the required logic.

Reply:
I don't have a strong opinion on how we solve it. But we should not explicitly need to hard code a operator name like SESSION. Otherwise we need to touch this part for every PTF we are adding.